File: app/src/main/java/com/yunho/nanobanana/domain/model/AIGenerationResult.kt
package com.yunho.nanobanana.domain.model

import android.graphics.Bitmap

/**
 * Domain model for AI generation results
 * Sealed class for type-safe result handling
 */
sealed class AIGenerationResult {
    /**
     * Successful generation with optional image and text
     */
    data class Success(
        val image: Bitmap? = null,
        val text: String? = null,
        val reasoning: String? = null
    ) : AIGenerationResult() {
        init {
            require(image != null || text != null) { 
                "At least one of image or text must be non-null" 
            }
        }
    }
    
    /**
     * Generation failed with error message
     */
    data class Error(
        val message: String,
        val throwable: Throwable? = null
    ) : AIGenerationResult()
    
    /**
     * Generation is in progress
     */
    data class Loading(
        val progress: Float = 0f,
        val message: String = "Generating...",
        val reasoning: String? = null
    ) : AIGenerationResult()
}

File: lib/models/ai_generation_result.dart
import 'dart:typed_data';

/// Domain model for AI generation results
/// Sealed classes for type-safe result handling
abstract class AIGenerationResult {
  const AIGenerationResult();
}

/// Successful generation with optional image and text
class AIGenerationSuccess extends AIGenerationResult {
  final Uint8List? image;
  final String? text;
  final String? reasoning;

  const AIGenerationSuccess({
    this.image,
    this.text,
    this.reasoning,
  }) : assert(
          image != null || text != null,
          'At least one of image or text must be non-null',
        );
}

/// Generation failed with error message
class AIGenerationError extends AIGenerationResult {
  final String message;
  final dynamic error;

  const AIGenerationError({
    required this.message,
    this.error,
  });
}

/// Generation is in progress
class AIGenerationLoading extends AIGenerationResult {
  final double progress;
  final String message;
  final String? reasoning;

  const AIGenerationLoading({
    this.progress = 0.0,
    this.message = 'Generating...',
    this.reasoning,
  });
}
